File: src/main/java/org/stt/text/TextModule.java
package org.stt.text;

import dagger.Module;
import dagger.Provides;

import java.util.Arrays;
import java.util.Collection;

@Module
public class TextModule {
    private TextModule() {
    }

    @Provides
    static ItemCategorizer provideItemCategorizer(WorktimeCategorizer worktimeCategorizer) {
        return worktimeCategorizer;
    }

    @Provides
    static Collection<ExpansionProvider> provideExpansionProvider(CommonPrefixGrouper commonPrefixGrouper,
                                                                  JiraExpansionProvider jiraExpansionProvider) {
        return Arrays.asList(commonPrefixGrouper, jiraExpansionProvider);
    }

    @Provides
    static ItemGrouper provideItemGrouper(CommonPrefixGrouper grouper) {
        return grouper;
    }
}

File: src/main/kotlin/org/stt/text/TextModule.kt
package org.stt.text

import dagger.Module
import dagger.Provides
import java.util.*

@Module
class TextModule {

    @Provides
    fun provideItemCategorizer(worktimeCategorizer: WorktimeCategorizer): ItemCategorizer {
        return worktimeCategorizer
    }

    @Provides
    fun provideExpansionProvider(commonPrefixGrouper: CommonPrefixGrouper,
                                 jiraExpansionProvider: JiraExpansionProvider): Collection<ExpansionProvider> {
        return Arrays.asList(commonPrefixGrouper, jiraExpansionProvider)
    }

    @Provides
    fun provideItemGrouper(grouper: CommonPrefixGrouper): ItemGrouper {
        return grouper
    }
}
